components/style/data.rs

```diff
@@ -193,6 +193,19 @@ impl fmt::Debug for EagerPseudoArray {
     }
 }
 
+#[cfg(feature = "gecko")]
+impl MallocSizeOfWithRepeats for EagerPseudoArray {
+    fn malloc_size_of_children(&self, state: &mut SizeOfState) -> usize {
+        let mut n = 0;
+        for i in 0..EAGER_PSEUDO_COUNT {
```

Comment:
I think this can just be `for val in self` (same for the `Display` impl.)

`for index in 0..n` is quite rare in Rust and usually indicates a loop over only a portion of the datastructure.